### PR TITLE
move kube-dns to the cluster/addons/ directory

### LINKED ISSUE #28553: maintaine kube-dns in cluster/addons
Not everyone uses salt. We need to maintain cluster addons (even required ones) in cluster/addons/ so they are visible to users. This is where people go to look for addons. The change in location is causing a lot of confusion for non-kube-up deployment users as there was no release notes or migration path on how to deploy from the new location.

cc @girishkalele

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #28553

cc @girishkalele @bprashanth

### CODE REVIEW

**Comment by girishkalele** [comment]
@mikedanese 

Need to fix all the references to saltbase/kube-dns/skydns-*.yaml - hyperkube, coreos are a couple that come to mind.

**Comment by girishkalele** [comment]
That looks like it covers every platform visible in our tree.
Probably need some pointers fixed in the kubernetes.io repo too.
Did you also grep in READMEs ?

I see a reference in build/kube-dns/README.md still there.

```
> Second, you need to start the DNS server ReplicationController and Service. See
> the example files ([ReplicationController](../../cluster/saltbase/salt/skydns-rc.yaml.in) and
> [Service](../../cluster/saltbase/salt/skydns-svc.yaml.in)), but keep in mind that these are templated for
```

File: cluster/stage-images.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (girishkalele) [inline cluster/stage-images.sh]: Not sure where this is used for DNS ?

File: build/kube-dns/README.md [hunk 2]
```diff
@@ -148,8 +148,8 @@ set:
 ```
 
 Second, you need to start the DNS server ReplicationController and Service. See
-the example files ([ReplicationController](../../cluster/saltbase/salt/skydns-rc.yaml.in) and
-[Service](../../cluster/saltbase/salt/skydns-svc.yaml.in)), but keep in mind that these are templated for
+the example files ([ReplicationController](../../cluster/addmons/dns/skydns-rc.yaml.in) and
```
Comment (girishkalele) [inline build/kube-dns/README.md]: typo ? ```addmons```

Comment (mikedanese) [inline build/kube-dns/README.md]: Fixed.

File: cluster/stage-images.sh (hunk 1 above)
Comment (mikedanese) [inline cluster/stage-images.sh]: Ooops did not mean to include this.

**Comment by girishkalele** [comment]
LGTM - should fix the majority of platforms bunding the DNS addon.

Minikube:
@puja108 - you mentioned minikube does not come with DNS - I haven't tried it myself, can you confirm ?

cc @luxas for hyperkube

**Comment by puja108** [comment]
@girishkalele not sure which version it was that didn't work for me. Tested with 0.5 and 0.6 again and it worked. However, there are no pods running just the service. The DNS itself seems to be running outside of the kubelets. Haven't looked too deep into it, yet.

**Comment by girishkalele** [comment]
That's quite interesting - the message "Automatic merge from submit-queue" was posted but then there was a conflict with the merge and it added a needs-rebase label.

**Comment by mikedanese** [comment]
I'm so confused. 7c7aa9835e0f4a79e2e2ee89fc242bfb43c75b3b

**Pull request merged**